TypeCobol/Compiler/Scanner/MultilineScanState.cs

```diff
@@ -371,8 +372,18 @@ private void AdjustPreviousTokenPropertiesBasedOnCurrentToken(Token newToken)
                             LastSignificantToken.DegradePotentialCodeElementStartingKeywordToSyntaxKeyword();
                         }
                         break;
+
+
+                    //DELETE_CD is the DELETE for Preprocessor
+                    //DELETE is for Cobol statement
+                    //Scanner cannot differentiate these 2 DELETE keywords 
+
+                    //DELETE_CD is valid only if followed by an integer or a range of number.
+                    //A range of number is scanned as an UserDefinedWord
+                    //A range of number contains only numeric character or char '-'
                     case TokenType.DELETE:
-                        if (newToken.TokenType == TokenType.IntegerLiteral)
+                        if (newToken.TokenType == TokenType.IntegerLiteral
+                            || (newToken.TokenType == TokenType.UserDefinedWord && newToken.Text.All(c=>Char.IsDigit(c) || c=='-')))
```

Comment:
missing spaces around '=>' and '=='

Reply:
Done in 77275cd8b6d33d153ae576f927f10f78b8e9591b